examples/rancher-gen/nginx.tmpl

```diff
@@ -89,30 +88,32 @@ server {
 ## TEMPLATES -------
 {{- define "upstream"}}
   {{- $serv := .}}
-  {{range $cnt := .Containers -}}
-    {{- $defaultPort := "80"}}
-    {{- $port := where $serv.Ports "InternalPort" $defaultPort }}
-    {{- if .Labels.Exists "rgon.port" }}
-      # Has rgon.port label
-      server {{$cnt.Address}}:{{$cnt.Labels.GetValue "rgon.port"}};
-    {{- else }}
-      # No rgon.port label
-      {{- $portLen := len $serv.Ports -}}
-      {{- if eq $portLen 0 }}
-      # No ports available
-      server {{$cnt.Address}} down;
-      {{- else }}
-      # Ports available
-        {{- range $p := $serv.Ports }}
-        server {{$cnt.Address}}:{{ $p.InternalPort }};
-        {{ end -}}
-      {{ end -}}
-    {{ end -}}
-  {{- end}}
-{{- end}}
+  {{- range $cnt := .Containers -}}
+    {{- $weight := $cnt.Labels.GetValue "rgon.weight" "1" }}
+    {{- $backup := $cnt.Labels.GetValue "rgon.backup" "" }}
+    {{- $portLen := len $serv.Ports }}
+    {{- $portInt := where $serv.Ports "InternalPort" "80" }}
+    {{- $portLabel := $cnt.Labels.GetValue "rgon.port" "" -}}
+
+    {{- if eq $cnt.State "stopped" }}{{ printf "\n  server %s down; # Container stopped" $cnt.Address }}
+    {{- else if (and (eq $portLen 0) (eq (len $portInt) 0 )) }}{{- printf "\n  server %s down; # No ports available" $cnt.Address -}}{{ end -}}
```

Comment:
should we really mark the container as down, or should we skip it?

Reply:
In these cases, I think it should be `down`.

A stopped container with an `rgon.domain` label is clearly intended to be accessible, and if it is down it is probably due to health issues or manual intervention.

A container with an `rgon.domain` label and no exposed ports is likely misconfigured, in which case this `down` parameter may help to provide a troubleshooting path.

Both cases do not impact healthy services.